File: 443. String Compression/GenSol.js
/**
 * @param {character[]} chars
 * @return {number}
 */
var compress = (chars) => {
    let res = 0
    let i = 0
    let letter = ""
    let count = 0

    while (i < chars.length) {
        
        letter = chars[i]
        count = 0
        
        while (i < chars.length && chars[i] === letter) {
            count += 1
            i += 1
        }

        chars[res] = letter
        res += 1

        if (count > 1) {
            for (const c of count.toString()) {
                chars[res] = c
                res += 1
            }
        }
    }
    return res
}

File: 443. String Compression/GenSol.cpp
class Solution
{
public:
    int compress(vector<char> &chars)
    {
        int res = 0;
        int i = 0;
        char letter = ' ';
        int count = 0;

        while (i < chars.size())
        {
            count = 0;
            letter = chars[i];

            while (i < chars.size() && chars[i] == letter)
            {
                i++;
                count++;
            }

            chars[res] = letter;
            res++;

            if (count > 1)
            {
                for (char c : std::to_string(count))
                {
                    chars[res] = c;
                    res++;
                }
            }
        }
        return res;
    }
};